app/views/layouts/application.haml

```diff
@@ -95,22 +95,22 @@
                 %li= link_to 'Favorites', favorites_path
                 - unless current_user.read_only?
                   %li= link_to "Post", new_post_path
-              = link_to logout_path, method: :delete, id: :"header-logout", data: { confirm: 'Press OK to confirm logout' } do
+              = link_to destroy_user_session_path, method: :delete, id: :"header-logout", data: { confirm: 'Press OK to confirm logout' } do
                 = button_tag "Log out", class: 'button'
-            - elsif !current_page?(login_path)
-              = form_tag login_path, method: :post, id: 'header-form' do
+            - elsif !current_page?(new_user_session_path)
+              = form_with scope: :user, url: user_session_path, id: 'header-form' do |f|
                 #header-forms
-                  = text_field_tag :username, params[:username], placeholder: "Username", id: 'header-username', autocomplete: 'username'
-                  = password_field_tag :password, params[:password], placeholder: "Password", id: 'header-password', autocomplete: 'current-password'
+                  = f.text_field :username, placeholder: "Username", id: 'header-username', autocomplete: 'username'
+                  = f.password_field :password, placeholder: "Password", id: 'header-password', autocomplete: 'current-password'
                 #header-remember
-                  = check_box_tag :remember_me, class: 'vmid', id: 'header-rememberme'
+                  = f.check_box :remember_me, class: 'vmid', id: 'header-rememberme'
                   = label_tag 'header-rememberme', "Remember Me", class: 'vmid'
```

Comment:
```suggestion
                  = f.check_box :remember_me, class: 'vmid', id: 'header-rememberme'
                  = f.label :remember_me, for: 'header-rememberme', id: 'header-rememberme', class: 'vmid'
``` seems like it should work? I checked the code and it does respect for and might even set the id from that, that part might be redundant.

Reply:
I think the ID is redundant! (Possibly invalid? Two fields with the same ID?)

Probably this:

```suggestion
                  = f.check_box :remember_me, class: 'vmid', id: 'header-rememberme'
                  = f.label :remember_me, for: 'header-rememberme', class: 'vmid'
```